components/kyma-environment-broker/cmd/cli/credential/manager.go

```diff
@@ -0,0 +1,147 @@
+package credential
+
+import (
+	"context"
+	"time"
+
+	"github.com/int128/kubelogin/pkg/adaptors/browser"
+	"github.com/int128/kubelogin/pkg/adaptors/certpool"
+	"github.com/int128/kubelogin/pkg/adaptors/clock"
+	"github.com/int128/kubelogin/pkg/adaptors/credentialpluginwriter"
+	"github.com/int128/kubelogin/pkg/adaptors/oidcclient"
+	"github.com/int128/kubelogin/pkg/adaptors/reader"
+	"github.com/int128/kubelogin/pkg/adaptors/tokencache"
+	"github.com/int128/kubelogin/pkg/usecases/authentication"
+	"github.com/int128/kubelogin/pkg/usecases/authentication/authcode"
+	"github.com/int128/kubelogin/pkg/usecases/authentication/ropc"
+	"github.com/int128/kubelogin/pkg/usecases/credentialplugin"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/cmd/cli/logger"
+	"golang.org/x/oauth2"
+	"k8s.io/client-go/util/homedir"
+)
+
+var defaultTokenCacheDir = homedir.HomeDir() + "/.kube/cache/oidc-login"
+var defaultListenAddress = []string{"127.0.0.1:8000", "127.0.0.1:18000"}
+
+const defaultAuthenticationTimeoutSec = 180
+
+// Manager is a client for an OIDC provider capable of authenticating users and retrieving ID tokens through
+//   - Authorization code grant flow using browser for interactive use
+//   - Resource owner password credentials flow for non-interactive use
+// Manager implements the oauth2.TokenSource interface to interact with client libraries depending on the oauth2 package for obtaining auth token.
+type Manager interface {
+	GetTokenByAuthCode(ctx context.Context) (string, error)
+	GetTokenByROPC(ctx context.Context, username, password string) (string, error)
+	TokenExpiry() time.Time
+	Token() (*oauth2.Token, error)
+}
+
+type manager struct {
```

Comment:
As I see, the manager struct is not thread-safe. For CLI it can be accepted, but please add a proper comment that it is not thread-safe (reusing input field).

Reply:
It is thread safe, because the only usage is copied to local variables inside the `GetTokenByAuthCode` and `GetTokenByROPC` methods. It is the local variable which gets modified inside the method bodies.

Reply:
ah, ok, fine